cpp/src/problems/longestConsecutiveSequence.cpp

```diff
@@ -0,0 +1,43 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+/**
+ * Given an unsorted array of integers array, return the length of the longest
+ * consecutive elements sequence
+ */
+
+//Example 1:
+//        Input: array = [100,4,200,1,3,2]
+//        Output: 4
+//        Explanation: The longest consecutive elements sequence is [1, 2, 3, 4]. Therefore its length is 4
+
+// Example 2:
+//        Input: array = [0,3,7,2,5,8,4,6,0,1]
+//        Output: 9
+
+class LongestConsecutive
+{
+public:
+    int longestConsecutive(vector<int> &array)
+    {
+        if (array.empty())
+            return 0;
+        unordered_set<int> record(array.begin(), array.end());
+        int longestSequence = 1;
+        for (int n : array)
```

Comment:
I like your idea to check for the record and run a while loop to purge those elements and keep moving forward or backward.
However, I find that the following things can be improved.
1. record.find() takes log(n) time in the worst case. It is not constant.
2. record.erase() should also ideally take log(n) since the element is again required to be searched for.
3. erasing an element may or may not (I am not of internal implementation) add extra time for reshuffling the elements. (I am not exactly sure).

Do you think you can take care of the first two points?